scripts/check_names.py

```diff
@@ -971,15 +971,17 @@ def parse_symbols(self):
             )
             my_environment = os.environ.copy()
             my_environment["CFLAGS"] = "-fno-asynchronous-unwind-tables"
-            # Run make clean separately to lib to prevent unwanted behavior when
-            # make is invoked with parallelism.
+
+            source_dir = os.getcwd()
+            build_dir = tempfile.mkdtemp()
+            os.chdir(build_dir)
             subprocess.run(
-                ["make", "clean"],
+                ["cmake", "-DGEN_FILES=ON", source_dir],
                 universal_newlines=True,
                 check=True
             )
             subprocess.run(
-                ["make", "lib"],
+                ["make"],
```

Comment:
Nit: `cmake --build .` would be slightly nicer as it allows someone to run this using a non-make backend (e.g. Ninja).